### ISSUE TITLE
How to set a correct random seed?

### ISSUE THREAD

**Issue by densechen** [pytorch/pytorch#45797]
Hi, I am using pytorch==1.3/1.1 with 4/2 GPUs to training network.
I use the following code to set random seed at the beginning of program:
`
def seed_torch(seed=1029):
    random.seed(seed)
    os.environ['PYTHONHASHSEED'] = str(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.cuda.manual_seed_all(seed)  # if you are using multi-GPU.
    torch.backends.cudnn.benchmark = False
    torch.backends.cudnn.deterministic = True

seed_torch()
`
The only random function I called during training/testing time is torch.randint.
However, I found that though I have set the random seed, the testing result is still different every time.
If I replace torch.randint with torch.zeros, I can get same accuracy even without setting the random seed.

I do not know why?
Can any one help me about this?

**Comment by VitalyFedyunin** [comment]
Hello! It is better to ask such kind of questions at https://discuss.pytorch.org/. I will close this issue, feel free to reopen if you think that you need to report a bug or make feature proposal. Thanks.
